File: DAY 2/removeAdjacentValues.c
#include<stdio.h>
#include<stdlib.h>
int stack[100000];
int top=-1;
void push(int a){
    stack[++top]=a;
}
int pop(){
    return stack[top--];
}
int isEmpty(){
    return top==-1;
}
int peek(){
    return stack[top];
}
int main()
{
    int a;
    scanf("%d",&a);
    int arr[a];
    for(int i=0;i<a;i++){
        scanf("%d",&arr[i]);
    }
    for(int i=a-1;i>=0;i--){
        if(isEmpty() || peek() != arr[i]){
            push(arr[i]);
        }
        else{
            pop();
        }
    }
    if(isEmpty()){
        printf("-1");
        return;
    }
    while(!isEmpty()){
        printf("%d ",pop());
    }
}

File: DAY 2/removeAdjacentVAlues.java
import java.util.*;

public class removeAdjacentVAlues {
    public static void main(String[] args) {
        int arr[] = { 20, 15, 10, 30, 30, 10, 15, 50, 90 };
        Stack<Integer> st = new Stack<Integer>();
        for (int i = arr.length - 1; i >= 0; i--) {
            if (st.isEmpty() || st.peek() != arr[i]) {
                st.push(arr[i]);
            } else {
                st.pop();
            }
        }
        if (st.isEmpty()) {
            System.out.println("-1");
            return;
        }
        while (!st.isEmpty()) {
            System.out.println(st.pop());
        }
    }
}
